dracut/image-boot/eos-image-boot-setup

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+#
+# Support booting from an image file hosted on a filesystem.
+# We identify which disk blocks correspond to the image file, and create
+# a dm-linear block device mapping them.
+
+type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+host_device=$(cat /var/tmp/endless-image-host)
+image_path=$(getarg endless.image.path)
+
+blockdev --setro ${host_device}
+
+if getargbool 0 endless.live_boot; then
+  dm_roflag=--readonly
+  kpartx_roflag=-r
+fi
+
+# Identify the EOS image extents on the host device, and create a dm-linear
+# block device that maps exactly to that.
+extents=$(dumpexfat -f "${image_path}" "${host_device}")
+if [ $? != 0 ]; then
+  echo "image-boot: failed to lookup image on device"
+  exit 1
+fi
+
+offset=0
+echo "$extents" | while read extent_offset extent_size; do
+  [ -z "${extent_offset}" -o -z "${extent_size}" ] && continue
+  # Convert bytes to sectors, rounding up
+  if [ $(( extent_size % 512 )) != 0 ]; then
+    echo "image-boot: extent size $extent_size not sector-aligned" >&2
+    exit 1
+  fi
+  echo "${offset} ${extent_size} linear ${host_device} $extent_offset"
```

Comment:
Since `dumpexfat` produces bytes, and the table format documented in `dmsetup(8)` is in terms of sectors, I think this line is now wrong: you need to divide `extent_size` and `extent_offset` by 512.